dali/operators/generic/slice/slice_attr.h

```diff
@@ -273,24 +339,55 @@ class SliceAttr {
   bool normalized_anchor_, normalized_shape_;
   std::vector<int> axes_;
   TensorLayout axis_names_;
+  std::vector<CropWindowGenerator> crop_window_generators_;
+};
 
-  ArgValue<int, 1> start_;
-  ArgValue<float, 1> rel_start_;
-
-  ArgValue<int, 1> end_;
-  ArgValue<float, 1> rel_end_;
-
-  ArgValue<int, 1> shape_;
-  ArgValue<float, 1> rel_shape_;
+class SliceAttr {
+ public:
+  explicit inline SliceAttr(const OpSpec &spec)
+      : spec_(spec),
+        named_slice_attr_(spec),
+        pos_slice_attr_(spec) {
+  }
 
-  std::vector<CropWindowGenerator> crop_window_generators_;
+  template <typename Backend>
+  void ProcessArguments(const workspace_t<Backend> &ws) {
+    use_named_args_ = named_slice_attr_.template ProcessArguments<Backend>(ws);
+    if (use_named_args_) {
+      if (spec_.HasArgument("normalized_anchor") || spec_.HasArgument("normalized_shape")) {
+        DALI_WARN(
+            "Warning: ``normalized_anchor``/``normalized_shape`` is only relevant "
+            "when using positional slice arguments");
+      }
+      DALI_ENFORCE(ws.NumInput() == spec_.GetSchema().MinNumInput(),
+                  "Named arguments start/end/shape are not compatible with positional"
+                  " anchor and shape inputs");
+    } else {
+      bool processed_pos_args = pos_slice_attr_.template ProcessArguments<Backend>(ws);
```

Comment:
The value returned here, compared to L355, means only 1 kind of error.
I think you should check the `if (ws.NumInput() != (spec_.GetSchema().MinNumInput() + 2))` explicitly here as in L362.

Reply:
done